src/Modules/OrchardCore.Commerce/Services/CardPaymentService.cs

```diff
@@ -0,0 +1,157 @@
+    private readonly ILogger<CardPaymentService> _logger;
+
+    public CardPaymentService(
+        IShoppingCartPersistence shoppingCartPersistence,
+        IPriceService priceService,
+        IPriceSelectionStrategy priceSelectionStrategy,
+        IContentManager contentManager,
+        ISiteService siteService,
+        IDataProtectionProvider dataProtectionProvider,
+        ILogger<CardPaymentService> logger)
+    {
+        _paymentIntentService = new PaymentIntentService();
+        _shoppingCartPersistence = shoppingCartPersistence;
+        _priceService = priceService;
+        _priceSelectionStrategy = priceSelectionStrategy;
+        _contentManager = contentManager;
+        _siteService = siteService;
+        _dataProtectionProvider = dataProtectionProvider;
+        _logger = logger;
+    }
+
+    public async Task<PaymentIntent> CreatePaymentAsync(ConfirmPaymentRequest request)
+    {
+        var currentShoppingCart = await _shoppingCartPersistence.RetrieveAsync();
+        var totals = await currentShoppingCart.CalculateTotalsAsync(_priceService, _priceSelectionStrategy);
+
+        // Same here as on the checkout page: Later we have to figure out what to do if there are multiple
+        // totals i.e., multiple currencies.
+        var defaultTotal = totals.FirstOrDefault();
+
+        var paymentIntent = new PaymentIntent();
+        var requestOptions =
+            new RequestOptions
+            {
+                ApiKey = (await _siteService.GetSiteSettingsAsync())
+                    .As<StripeApiSettings>()
+                    .SecretKey
+                    .DecryptStripeApiKey(_dataProtectionProvider, _logger),
+            };
+
+        if (request.PaymentMethodId != null)
+        {
+            var paymentIntentOptions = new PaymentIntentCreateOptions
+            {
+                // We need to remove the decimal points and convert the value (decimal) to long.
+                // https://stripe.com/docs/currencies#zero-decimal
+                Amount = long
+                .Parse(
+                    string.Join(
+                        string.Empty,
+                        defaultTotal.ToString().Where(char.IsDigit)),
+                    CultureInfo.InvariantCulture),
+                Currency = defaultTotal.Currency.CurrencyIsoCode,
+                Description = "Orchard Commerce Test Stripe Card Payment",
+                ConfirmationMethod = "manual",
+                Confirm = true,
+                PaymentMethod = request.PaymentMethodId,
+
+                // If shipping is implemented, it needs to be added here too.
+                // Shipping =
+                // ReceiptEmail = viewModel.Email,
+            };
+
+            paymentIntent = _paymentIntentService.Create(paymentIntentOptions, requestOptions);
+        }
+
+        if (request.PaymentIntentId != null)
+        {
+            paymentIntent = _paymentIntentService.Confirm(
```

Comment:
Rider says these have async counterparts. Are you intentionally avoiding them? If so, please include a comment.
![image](https://user-images.githubusercontent.com/4007293/181866185-75cdb759-0010-4174-8c0d-1c80ca67a992.png)

Reply:
Hmm. I didn't get any warning. I'm using the not `async` version because this was in the documentation too. https://stripe.com/docs/payments/accept-a-payment-synchronously?platform=web#create-payment-intent (line 46)

Reply:
They were probably cheaping out on documentation, so they don't have to show off both versions. Please check if it works with the async versions, usually it's not a good idea to synchronously wait for a web request.